### DISCUSSION TITLE
What is the canonical way to implement a states stack patern?

### DISCUSSION THREAD

**Discussion by DigammaF** [pokepetter/ursina#329] [Q&A]
I want different states for my game: main menu, game menu, game, options menu ect. Should I use different Camera objects, different Scene objects or different Application objects even? (I greatly doubt about the last one though).

**Comment by pokepetter** [comment]
Use entities, parent other entities to them and set .enabled a you wish. If there's many states and you only want one enabled at a time, you could use the Animator for that.

**Reply by DigammaF** [reply]
Thank you! So I should, for instance, use an Entity called MainMenu and enable/disable it at will.
